iacrcc/iacrcc.cls

```diff
@@ -144,72 +144,112 @@
+    \RequirePackage{hyperref}
+  \fi
+  %
+  % Pass the keeppdfinfo option to hyperxmp to ensure the
+  % Info dictionary is retained which contain the author names
+  \hypersetup{keeppdfinfo}
   \if@anonymous
     \hypersetup{pdfauthor={hidden for submission}}
   \else
-    \hypersetup{pdfauthor=\IACR@listofauthors}
+    \hypersetup{pdfauthor={\IACR@listofauthors}}
   \fi
-  \hypersetup{pdftitle={\@plaintitle}}
-  \if\@IACR@title@subtitle\empty
-  \else
+  \ifdefined\@IACR@title@subtitle@defined
     \hypersetup{pdfsubtitle={\@IACR@title@subtitle}}
   \fi
-  % other PDF metadata
+  \ifx\IACR@text@keywords\@empty\else
+    \hypersetup{pdfkeywords={\IACR@text@keywords}}
+  \fi
+  \hypersetup{pdftitle={\@plaintitle},%
+              pdflang=en,%
+              urlcolor=blue,%
+              colorlinks=true,%
+              citecolor=black!70!green,%
+              linkcolor=black!70!red%
+             }
   \ifcsstring{@IACRversion}{final}{%
     \hypersetup{%
       pdfpubstatus=VoR,
-      pdfdoi=\IACR@DOI,%
-      pdfissn=\IACR@ISSN,%
+      pdfdoi={\IACR@DOI},%
+      pdfissn={\IACR@ISSN},%
       pdfpubtype=journal,%
       pdfpublication={\publname},%
       pdfvolumenum={\IACR@vol},%
       pdfissuenum={\IACR@no},%
-      pdfpagerange={\IACR@fp-\IACR@lp},%
+      pdflicenseurl={\doclicenseURL},%
+      pdfsubject={\publname{}, DOI:\IACR@DOI}%
     }
+    % doclicense added support for the macro \doclicenseLongTextForHyperref in version 3.1.0 (2022/04/27)
+    \if@doclicense@LongTextForHyperref
+      \hypersetup{pdfcopyright={\doclicenseLongTextForHyperref}}
+    \else
+      \hypersetup{pdfcopyright={This work is licensed under a \doclicenseLongName\space license.}}
+    \fi
+  }{}% final
+  %
+  % Disable latexdiff commands in PDF links
+  \pdfstringdefDisableCommands{%
+    \def\DIFadd#1{#1}%
+    \def\DIFdel#1{}%
+  }
+  % The hyperref package is used to handle cross-referencing commands in
+  % LaTeX to produce hypertext links in the document. hyperref is fussy about
+  % the order in which it is loaded, and the official documentation recommends
+  % loading last. In spite of this, there are some packages that must be
+  % loaded after hyperref. (e.g., cleveref). In order to support this, the
+  % author should create a file called after-hyperref.sty with
+  % \RequirePackage{cleveref} in it.
+  \IfFileExists{after-hyperref.sty}{%
+    \RequirePackage{after-hyperref}
+  }{}
+  \ifcsstring{@IACRversion}{final}{%
+    % When compiling the final version we need the last page number
+    % totpages: count pages in a document, and report last page number
+    % The documentation recommends to load this package last.
+    \RequirePackage{totpages}
+  }{}% final
+}
+
+% The TotPages label is not defined until after the \begin{document}.
+\AtBeginDocument{%
+  \ifcsstring{@IACRversion}{final}{%
+    \gdef\IACR@lp{\ref*{TotPages}}
+    \hypersetup{%
+      pdfnumpages={\IACR@lp},%
+      pdfpagerange={\IACR@fp-\IACR@lp},%
```

Comment:
I think that `\IACR@lp` is now being set correctly (based on looking at test14).